### PR TITLE
Player turn tests

### PR DESCRIPTION
**Pull request by JefeNelson**
After days of fighting with this... Still has bugs, unfortunately.

I pulled Chelsea's recent code to see if I'm getting the issues with the bishop, because it was bothering me with testing of player turn. And it works. :) So this may have to be merged after Chelsea's pull request gets merged.

**Anyways, the issues I am facing now is:**
-When a game is first started, white gets to move twice. After that, the code works well.
-If the page is refreshed with the refresh button, the current player gets to move twice.

I'm trying to figure out what happens after a refresh and what's causing it so any feedback is appreciated.

Also Smith, I have altered your 'switch_player_turn' method as it wasn't functioning right for me. You can switch it back if it works.

**Far from complete.** 

Also, the huge current turn display will be polished at the end. In place for testing.

### CODE REVIEW

**Comment by JefeNelson** [comment]
I realize the changes will be confusing as it includes changes from the 'prevent_moving_into_check'. Sorry, I needed the bishop to work.

**Comment by smiths15** [comment]
@JefeNelson getting same error with player turn. I will keep looking into it and let you know if I find the bug

**Comment by JefeNelson** [comment]
Issues fixed. :-)

Feedback welcome.

Thank you, Chelsea.. for fixing that bishop bug.

File: app/views/games/show.html.erb [hunk 1]
```diff
@@ -49,6 +48,9 @@ function init() {
 };
 
 $( init );
+$(document).ajaxStop(function(){
```
Comment (smiths15) [inline app/views/games/show.html.erb:51]: works great! 👍

**Review by smiths15 — APPROVED** [review]

**Pull request merged**